README.md

```diff
@@ -130,57 +105,72 @@ NVSentinel follows a microservices architecture with modular health monitors and
-        PC["Platform Connectors"]
-        STORE[("MongoDB Store<br/>Event Database")]
-        FQ["Fault Quarantine"]
-        ND["Node Drainer"]
-        FR["Fault Remediation"]
-        HEA["Health Events Analyzer"]
+        PC["Platform Connectors<br/>(gRPC Server)"]
+        STORE[("MongoDB Store<br/>(Event Database)")]
+        FQ["Fault Quarantine<br/>(Node Cordon)"]
+        ND["Node Drainer<br/>(Workload Eviction)"]
+        FR["Fault Remediation<br/>(Break-Fix Integration)"]
+        HEA["Health Events Analyzer<br/>(Pattern Analysis)"]
+        LBL["Labeler<br/>(Node Labels)"]
     end
     
-    GPU --> PC
-    SYS --> PC
-    CSP --> PC
-
+    subgraph "Kubernetes Cluster"
+        K8S["Kubernetes API<br/>(Nodes, Pods, Events)"]
+    end
+    
+    GPU -->|gRPC| PC
+    SYS -->|gRPC| PC
+    CSP -->|gRPC| PC
+    
+    PC -->|persist| STORE
+    PC <-->|update status| K8S
     
-    PC --> STORE
+    FQ -.->|watch changes| STORE
+    FQ -->|cordon| K8S
     
-    FQ -.->|watches| STORE
-    ND -.->|watches| STORE
-    FR -.->|watches| STORE
-    HEA -.->|watches| STORE
+    ND -.->|watch changes| STORE
+    ND -->|drain| K8S
+    
+    FR -.->|watch changes| STORE
+    FR -->|create CRDs| K8S
+    
+    HEA -.->|watch changes| STORE
+    
+    LBL -->|update labels| K8S
 ```
 
 **Data Flow**:
-1. **Health Monitors** detect hardware/software faults and send events via gRPC
-2. **Platform Connectors** receive and persist events to MongoDB
-3. **Core Modules** independently watch MongoDB for relevant events via change streams
-4. **Each module** acts autonomously based on their configured rules and policies
+1. **Health Monitors** detect hardware/software faults and send events via gRPC to Platform Connectors
+2. **Platform Connectors** validate, persist events to MongoDB, and update Kubernetes node conditions
+3. **Core Modules** independently watch MongoDB change streams for relevant events
+4. **Modules** interact with Kubernetes API to cordon, drain, label nodes, and create remediation CRDs
+5. **Labeler** monitors pods to automatically label nodes with DCGM and driver versions
 
-> **Note**: Modules operate independently and don't communicate directly with each other. All coordination happens through the shared MongoDB event store using change streams.
+> **Note**: All modules operate independently without direct communication. Coordination happens through MongoDB change streams and Kubernetes API.
 
 ## ⚙️ Configuration
 
-### Quick Configuration
+### Global Settings
 
-Enable/disable modules and set global options:
+Control module enablement and behavior:
 
 ```yaml
 global:
-  # Global settings
-  dryRun: false  # Enable for testing without actual actions
+  dryRun: false  # Test mode - no actual actions
   
   # Health Monitors (enabled by default)
   gpuHealthMonitor:
     enabled: true
   syslogHealthMonitor:
     enabled: true
+  cspHealthMonitor:
+    enabled: false  # Cloud provider integration
```

Comment:
the heading above says `# Health Monitors (enabled by default)`, but this is disabled, should add an empty line to make it obvious, what do you think?